src/magic.cpp

```diff
@@ -1686,25 +1686,25 @@ void known_magic::learn_spell( const spell_type *sp, Character &guy, bool force
             for( const trait_id &cancel : sp->spell_class->cancels ) {
                 if( cancel == sp->spell_class->cancels.back() &&
                     sp->spell_class->cancels.back() != sp->spell_class->cancels.front() ) {
-                    trait_cancel = string_format( _( "%s and %s" ), trait_cancel, cancel->name() );
+                    trait_cancel = string_format( _( "%s and %s" ), trait_cancel, cancel->name( "" ) );
                 } else if( cancel == sp->spell_class->cancels.front() ) {
-                    trait_cancel = cancel->name();
+                    trait_cancel = cancel->name( "" );
```

Comment:
Have you considered adding an overload without any parameters for the `name` function (and `desc`) instead?

As a reader I would know why this uses an empty string here. A call without any parameters would not raise such a question.

Reply:
The parameter is required because when I was first implementing it, there were a lot of calls to this, and I wanted to ensure that every call to name had to at least consider using the correct variant. My approach has changed a little since then, and there are a lot less calls to this function, so it's probably worth having that overload/to default the argument.